include/ck_tile/host/convolution_host_tensor_descriptor_helper.hpp

```diff
@@ -0,0 +1,351 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck_tile/ops/common/tensor_layout.hpp"
+#include "ck_tile/host/convolution_parameter.hpp"
+#include "ck_tile/host/host_tensor.hpp"
+
+namespace ck_tile {
+namespace conv {
+namespace detail {
+
+template <typename OldLayout>
+CK_TILE_HOST std::vector<std::size_t> get_layout_transpose_gnchw_to_old()
+{
+    // HACK: NHWC/KYXC/NHWK, which is treated as GNHWC/GKYXC/GNHWK by this function,
+    // is used by some legacy kernel. New kernel should use GNHWK/GKYXC/GNHWK
+    // TODO: remove this branch after removing legacy kernel
+    if constexpr(std::is_same_v<OldLayout, ck_tile::tensor_layout::convolution::NWC> ||
+                 std::is_same_v<OldLayout, ck_tile::tensor_layout::convolution::KXC> ||
+                 std::is_same_v<OldLayout, ck_tile::tensor_layout::convolution::NWK>)
+    {
+        return {0, 1, 3, 2};
+    }
```

Comment:
Maybe we should remove this hack ? :) We probably won't support mentioned "legacy" kernel in ck_tile.